src/context/TodoContext.tsx

```diff
@@ -0,0 +1,213 @@
+import React, {
+
+  RefObject,
+  SetStateAction,
+  useCallback, useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import {
+  createTodo, deleteTodo, getTodos, patchTodo,
+} from '../api/todos';
+import { ErrorMessages } from '../types/ErrorMessages';
+import { Filters } from '../types/Filters';
+
+const USER_ID = 11208;
+
+type ContextValue = {
+  filteredTodos: Todo[];
+  todos: Todo[];
+  loadingTodoId: number[];
+  tempTodo: Todo | null;
+  error: string | null;
+  filter: Filters;
+  nrOfActiveTodos: number;
+  completedTodosId: number[];
+  inputRef: RefObject<HTMLInputElement>;
+  addTodo: (todo: string) => void;
+  removeTodo: (id: number) => void;
+  updateTodo: (itemId: number, completed: boolean) => void;
+  removeAllCompleted: () => void;
+  changeErrorMessage: (message: string) => void;
+  setError: React.Dispatch<SetStateAction<string | null>>;
+  toggleAll: () => void;
+  setFilter: React.Dispatch<SetStateAction<Filters>>;
+};
+
+export const TodoContext = React.createContext<ContextValue | null>(null);
```

Comment:
Not a mistake here, but as a personal opinion: not a fan of creating a context for such a purpose. 
It may work for a small application, but may quicky become quite painful as the app grows. 

As an alternative for hooks: 
https://www.youtube.com/watch?v=SO8lBVWF2Y8

More on useContext: 
https://medium.com/@msgold/using-usecontext-in-react-a-comprehensive-guide-8a9f5271f7a8